### PR TITLE
fix(Core/Client Crash) Added diff showed in #2345 to prevent exploit crashes

### LINKED ISSUE #2170: Exploit Crash spell ( Malformed Packet ) [ $20 bounty ]
hello
i have a crash on my source , that seems to be an exploit 
this is my crashlogs 
https://gist.github.com/blackdev01/827699f9682ff12155cf770530e09623
https://gist.github.com/blackdev01/88d91dfedfbf789f2e847f7966c7a2a6
https://gist.github.com/blackdev01/fafda762ee8e2fc1a15bb269740fe906
https://gist.github.com/blackdev01/5f7b2cd3d4191227b06027ed257f1ea7

this my rev :  https://github.com/azerothcore/azerothcore-wotlk/commit/a9b981d619c220d7459a963cac4e989215b638b1
os : debian 8
i saw an report about this problem , i think my problem is look like this one 

this post is a bounty post i'll pay 20$ for this
my Discord : ProGrammer#8649

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/78174280-exploit-crash-spell-malformed-packet-20-bounty?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2198: Expolit crash
https://pastebin.com/7DH1ghZB - crashlog

Since there is no reliable information on this exploit, I decided to open new issues with all the information I know

The hacker uses the program, just go to the server to press the button and crash happens, unfortunately I do not have a program

"There is no need to carry out any specific actions; just press 1 button"

There are suspicions that the problem is here "Map.cpp - void Map :: Update"

This crash also works on the latest version of TC and  AC

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/78669930-expolit-crash?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Helias**
##### CHANGES PROPOSED:
-  diff from #2345

##### ISSUES ADDRESSED:
- close #2198
- close #2170
- close #2198
- related to #2043

##### TESTS PERFORMED:
- I didn't test it.

##### HOW TO TEST THE CHANGES:
 - send malformed packet to the wow server

##### Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by avatar789** [comment]
Does not work
crashlog https://pastebin.com/ecY2PYtV

**Comment by Helias** [comment]
Thank you for the feedback, could you tell me what are you using to test it?

**Comment by pklloveyou** [comment]
```c++
	}

	//events are executed in the same update
    if (unit->GetVehicle())
		unit->ExitVehicle();

	seat->second.Passenger.Guid = unit->GetGUID();
	seat->second.Passenger.IsUnselectable = unit->HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
```
Like this?

File: src/server/game/Chat/ChatLink.cpp [hunk 1]
```diff
@@ -661,7 +661,7 @@ bool LinkExtractor::IsValidMessage()
             break;
 
         char commandChar;
-        _iss >> commandChar;
```
Comment (pklloveyou) [inline src/server/game/Chat/ChatLink.cpp]: There should be no need to delete

**Comment by avatar789** [comment]
this software https://wowjp.net/forum/187-286970-1

File: src/server/game/Chat/ChatLink.cpp (hunk 1 above)
Comment (Helias) [reply src/server/game/Chat/ChatLink.cpp]: _iss >> commandChar is the same to _iss.get(commandChar) ?

**Comment by pklloveyou** [comment]
https://vk.com/market-173487801

**Comment by Helias** [comment]
I'm not able to get this software/addon, is there any other way to test this..?

**Comment by Helias** [comment]
PR moved here: https://github.com/azerothcore/azerothcore-wotlk/pull/2407
test it if you can :pray:

**Pull request closed without merging**